### ISSUE TITLE
Problem output is intermittent

### ISSUE THREAD

**Issue by bndbsh** [rust-lang/vscode-rust#330]
Sometimes, the problem output just doesn't show up. If I make a few dummy edits, it may reappear again. Then it may disappear on a given edit. I haven't noticed any pattern, it just seems like it fails sometimes due to some kind of race condition or intermittent error.

A cargo clean returns to normal behavior for a while, but eventually it happens again.

**Comment by mninja** [comment]
Some additional information: usually there's a spinner in the status bar when rls is doing work. I noticed that before the output disappears, there is no activity in the status bar for a couple of seconds. 

Launching the extension via vscode debug didn't yield any useful output.

**Comment by Xanewok** [comment]
I'm not sure I understand this. Do you mean that diagnostics just disappear from the Problems list and reappear again? Could you please provide at least some reproduction for this issue?

**Comment by mninja** [comment]
Yes, the diagnostics would just disappear. It happened consistently on any project for me, though lately I haven't been able to install rls to see if it's still happening.
